lib/node_modules/@stdlib/array/base/take-map/lib/assign.js

```diff
@@ -0,0 +1,273 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isComplexDataType = require( '@stdlib/array/base/assert/is-complex-floating-point-data-type' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex' );
+var ind = require( '@stdlib/ndarray/base/ind' ).factory;
+
+
+// FUNCTIONS //
+
+/**
+* Takes elements from an indexed array and assigns the values to elements in an indexed output array.
+*
+* @private
+* @param {Collection} x - Input array.
+* @param {IntegerArray} indices - List of indices.
+* @param {string} mode - Index mode.
+* @param {Collection} out - Output array.
+* @param {integer} stride - Output array stride.
+* @param {NonNegativeInteger} offset - Output array offset.
+* @param {Function} clbk - Callback function applied to each selected element.
+* @returns {Collection} Output array.
+*
+* @example
+* var x = [ 1, 2, 3, 4 ];
+* var indices = [ 3, 1, 2, 0 ];
+* var out = [ 0, 0, 0, 0 ];
+*
+* var arr = takeMapIndexed( x, indices, 'throw', out, 1, 0, function(val) {
+*   return val;
+* });
```

Comment:
I'll suggest to take declare `clbk` seperately and pass it as a parameter, that way it looks clean.

Reply:
cc @vr-varad